pkg/resourcemanager/vm/reconcile.go

```diff
@@ -52,9 +52,14 @@ func (g *AzureVirtualMachineClient) Ensure(ctx context.Context, obj runtime.Obje
 	// to overcome the issue with the lack of idempotence of the Create call
 	item, err := g.GetVirtualMachine(ctx, resourceGroup, resourceName)
 	if err == nil {
-		instance.Status.Provisioned = true
-		instance.Status.Provisioning = false
-		instance.Status.Message = resourcemanager.SuccessMsg
+		if *item.ProvisioningState == "Succeeded" {
```

Comment:
Is there an enum or const string you can use for the ```"Succeeded"``` string comparison?

Reply:
yes i can fix that